### PR TITLE
Self serve for GraphAPI compute

### PR DESCRIPTION
**Pull request by MehaKaushik**
[Preview this branch](https://cosmos-explorer-preview.azurewebsites.net/pull/985?feature.graphDedicatedGateway=true)

### CODE REVIEW

File: src/SelfServe/Example/SelfServeExample.rp.ts [hunk 1]
```diff
@@ -68,4 +68,4 @@ export const onRefreshSelfServeExample = async (): Promise<RefreshResult> => {
     isUpdateInProgress: progressToBeSent,
     updateInProgressMessageTKey: "UpdateInProgressMessage",
   };
-};
```
Comment (srinathnarayanan) [inline src/SelfServe/Example/SelfServeExample.rp.ts:71]: undo this change

File: tsconfig.strict.json [hunk 2]
```diff
@@ -95,6 +95,7 @@
     "./src/SelfServe/Example/SelfServeExample.types.ts",
     "./src/SelfServe/SelfServeStyles.tsx",
     "./src/SelfServe/SqlX/SqlxTypes.ts",
+    "./src/SelfServe/GraphDedicatedGateway/GraphDedicatedGatewayTypes.ts",
```
Comment (srinathnarayanan) [inline tsconfig.strict.json:98]: you dont need to add it here

File: src/SelfServe/Example/SelfServeExample.rp.ts (hunk 1 above)
Comment (MehaKaushik) [reply src/SelfServe/Example/SelfServeExample.rp.ts]: done

File: src/Localization/en/GraphDedicatedGateway.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
+{
+    "GraphAPIDescription": "Provision a Graph Dedicated Gateway for your Azure Cosmos DB account.",
```
[the remark this answers is not in the document]
Comment (MehaKaushik) [reply src/Localization/en/GraphDedicatedGateway.json:2]: Can you elaborate ?

File: src/SelfServe/GraphDedicatedGateway/GraphDedicatedGateway.rp.ts [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+  return `/subscriptions/${subscriptionId}/resourceGroups/${resourceGroup}/providers/Microsoft.DocumentDB/databaseAccounts/${name}/services/GremlinV2`;
+};
+
+export const updateDedicatedGatewayResource = async (sku: string, instances: number): Promise<string> => {
+  const path = getPath(userContext.subscriptionId, userContext.resourceGroup, userContext.databaseAccount.name);
+  const body: UpdateDedicatedGatewayRequestParameters = {
+    properties: {
+      instanceSize: sku,
+      instanceCount: instances,
+      serviceType: "GremlinV2",
```
[the remark this answers is not in the document]
Comment (MehaKaushik) [reply src/SelfServe/GraphDedicatedGateway/GraphDedicatedGateway.rp.ts:40]: added constant in this file itself

**Comment by MehaKaushik** [comment]
Moved here https://github.com/Azure/cosmos-explorer/pull/1005

File: src/SelfServe/GraphDedicatedGateway/GraphDedicatedGateway.rp.ts [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+  } catch (err) {
+    return new Array<string>();
+  }
+};
+
+const getFetchPricesPathForRegion = (subscriptionId: string): string => {
+  return `/subscriptions/${subscriptionId}/providers/Microsoft.CostManagement/fetchPrices`;
+};
+
+export const getPriceMap = async (regions: Array<string>): Promise<Map<string, Map<string, number>>> => {
```
[the remark this answers is not in the document]
Comment (MehaKaushik) [reply src/SelfServe/GraphDedicatedGateway/GraphDedicatedGateway.rp.ts:169]: no, the  billing story is not ready yet. I am using sqlx's pricemap for time being. Would update once billing is finalized.
